pkg/controller/workloadspread/reschedule.go

```diff
@@ -0,0 +1,137 @@
+	if oldCondition == nil {
+		if unschedulable {
+			SetWorkloadSpreadSubsetCondition(subsetStatus, NewWorkloadSpreadSubsetCondition(appsv1alpha1.SubsetSchedulable, corev1.ConditionFalse, "", ""))
+		} else {
+			SetWorkloadSpreadSubsetCondition(subsetStatus, NewWorkloadSpreadSubsetCondition(appsv1alpha1.SubsetSchedulable, corev1.ConditionTrue, "", ""))
+		}
+		return scheduleFailedPods
+	}
+	// initialize to old condition.
+	SetWorkloadSpreadSubsetCondition(subsetStatus, oldCondition.DeepCopy())
+
+	if unschedulable {
+		SetWorkloadSpreadSubsetCondition(subsetStatus, NewWorkloadSpreadSubsetCondition(appsv1alpha1.SubsetSchedulable, corev1.ConditionFalse, "", ""))
+	} else {
+		if oldCondition.Status == corev1.ConditionFalse {
+			expectReschedule := oldCondition.LastTransitionTime.Add(ScheduledFailedDuration)
+			currentTime := time.Now()
+			// the duration of unschedule status more than 5 minutes, recover to schedulable.
+			if expectReschedule.Before(currentTime) {
+				r.recorder.Eventf(ws, corev1.EventTypeNormal,
+					"RecoverSchedulable", "Subset %s of WorkloadSpread %s/%s is recovered from unschedulable to schedulable",
+					subsetStatus.Name, ws.Namespace, ws.Name)
+				SetWorkloadSpreadSubsetCondition(subsetStatus, NewWorkloadSpreadSubsetCondition(appsv1alpha1.SubsetSchedulable, corev1.ConditionTrue, "", ""))
+			} else {
+				// less 5 minutes, keep unschedulable.
+				durationStore.Push(getWorkloadSpreadKey(ws), expectReschedule.Sub(currentTime))
+			}
+		}
+	}
+
+	return scheduleFailedPods
+}
+
+func (r *ReconcileWorkloadSpread) cleanupUnscheduledPods(ws *appsv1alpha1.WorkloadSpread,
+	scheduleFailedPodsMap map[string][]*corev1.Pod) error {
+	for subsetName, pods := range scheduleFailedPodsMap {
+		if err := r.deletePodsForSubset(ws, pods, subsetName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ReconcileWorkloadSpread) deletePodsForSubset(ws *appsv1alpha1.WorkloadSpread,
+	pods []*corev1.Pod, subsetName string) error {
+	for _, pod := range pods {
+		if err := r.Client.Delete(context.TODO(), pod); err != nil {
+			r.recorder.Eventf(ws, corev1.EventTypeWarning,
+				"DeletePodFailed",
+				"Failed to delete unschedulabe Pod %s/%s in Subset %s of WorkloadSpread %s/%s",
+				pod.Namespace, pod.Name, subsetName, ws.Namespace, ws.Name)
+			return err
+		}
+		klog.V(3).Infof("WorkloadSpread (%s/%s) delete unschedulabe Pod (%s/%s) in Subset %s successfully",
+			ws.Namespace, ws.Name, pod.Namespace, pod.Name, subsetName)
+	}
+	return nil
+}
+
+// PodUnscheduledTimeout return true when Pod was scheduled failed and timeout.
+func PodUnscheduledTimeout(ws *appsv1alpha1.WorkloadSpread, pod *corev1.Pod) bool {
+	if pod.DeletionTimestamp != nil || pod.Spec.NodeName != "" || pod.Status.Phase != corev1.PodPending {
+		return false
+	}
+	for _, condition := range pod.Status.Conditions {
+		if condition.Type == corev1.PodScheduled && condition.Status == corev1.ConditionFalse &&
```

Comment:
if condition.Type == corev1.PodScheduled && condition.Status == corev1.ConditionFalse { } ?

Reply:
Perhaps the first two conditions are enough.